objectivec/generate_well_known_types.sh

```diff
@@ -53,24 +54,36 @@ declare -a RUNTIME_PROTO_FILES=( \
   google/protobuf/type.proto \
   google/protobuf/wrappers.proto)
 
+declare -a OBJC_EXTENSIONS=( .pbobjc.h .pbobjc.m )
+
 # Generate to a temp directory to see if they match.
 TMP_DIR=$(mktemp -d)
 trap "rm -rf ${TMP_DIR}" EXIT
 ./protoc --objc_out="${TMP_DIR}" ${RUNTIME_PROTO_FILES[@]}
-set +e
-diff -r "${TMP_DIR}/google" "${ProtoRootDir}/objectivec/google" > /dev/null
-if [[ $? -eq 0 ]] ; then
+
+DID_COPY=0
+for PROTO_FILE in "${RUNTIME_PROTO_FILES[@]}"; do
+  DIR=${PROTO_FILE%/*}
+  BASE_NAME=${PROTO_FILE##*/}
+  # Drop the extension
+  BASE_NAME=${BASE_NAME%.*}
+  OBJC_NAME=$(echo "${BASE_NAME}" | awk -F _ '{for(i=1; i<=NF; i++) printf "%s", toupper(substr($i,1,1)) substr($i,2);}')
+
+  for EXT in "${OBJC_EXTENSIONS[@]}"; do
+    if ! diff "${ObjCDir}/GPB${OBJC_NAME}${EXT}" "${TMP_DIR}/${DIR}/${OBJC_NAME}${EXT}" > /dev/null 2>&1 ; then
+      if [[ "${CHECK_ONLY}" == 1 ]] ; then
+        echo "ERROR: The WKTs need to be regenerated! Run $0"
```

Comment:
Can you change WKT to WellKnownTypes just so that I don't have to remember the acronym?

Reply:
Given the line ends with the script to run, which spells out what is being generated, I don't think it is needed here.